bundles/org.openhab.core.storage.json/src/main/java/org/openhab/core/storage/json/internal/JsonStorage.java

```diff
@@ -97,7 +97,7 @@ public JsonStorage(File file, @Nullable ClassLoader classLoader, int maxBackupFi
         commitTimer = new Timer();
 
         Map<String, StorageEntry> inputMap = null;
-        if (file.exists()) {
```

Comment:
Thanks for this draft. Imo this check should be moved into the `readDatabase(File)` method. When doing so we can additionally handle a 0 size backup file. Please also add a log warning with an explaining message to the user.